falcon/hooks.py

```diff
@@ -102,6 +115,13 @@ def after(action, *args, **kwargs):
             order given, immediately following the *req*, *resp*, *resource*,
             and *params* arguments.
 
+    Keyword Args:
+        is_async (bool): Set to ``True`` for ASGI apps to provide a hint that
+            the decorated responder is a coroutine function (i.e., that it
+            is defined with ``async def``). This is only necessary for
```

Comment:
Hmm... maybe we could tone down language here somewhat :thinking: 
I'm thinking of the example that Cython's Stefan provided in a related discussion, something along the lines of
```python
>>> import asyncio
>>> async def test():
...     print('Hello!')
...     await asyncio.sleep(1)
... 
>>> def coro_factory():
...     return test()
... 
>>> async def main():
...     await coro_factory()
... 
>>> asyncio.run(main())
Hello!
>>> import inspect
>>> inspect.iscoroutinefunction(test)
True
>>> inspect.iscoroutinefunction(coro_factory)
False
```
TL;DR you can never _really_ know whether a function is a coroutine function (except, of course, trying to call it and see).

Reply:
While technically it is true that you can't know without calling in all cases, usually you are really talking about something that was declared via async def, so you could argue that what we really want to know is "was this function declared with async def" (i.e., you are making an equivalency argument between coroutine functions and the syntax typically used to declare them).

Reply:
I suppose the factory pattern is probably the most likely exception to this rule... Let me see about reworking these is_async docstrings.

Reply:
Ah, that's true, you are actually defining it that way.
In any case, what I wanted to say was that this `is_async` flag might be useful in any kind of edge case, not just cythonized functions.

Reply:
Done. Let me know what you think.